src/Swarm/TUI/Controller.hs

```diff
@@ -590,25 +612,49 @@ handleREPLEvent s (Key V.KUp) =
   continue $ s & adjReplHistIndex Older
 handleREPLEvent s (Key V.KDown) =
   continue $ s & adjReplHistIndex Newer
+handleREPLEvent s (ControlKey 'r') = continue $
```

Comment:
pressing <kbd>CTRL-R</kbd> should scroll back in the history, to access previous matching entry.